src/js/components/Button/Button.js

```diff
@@ -37,22 +45,24 @@ class Button extends Component {
       label,
       onClick,
       plain,
+      primary,
       reverse,
       theme,
       type,
       ...rest
     } = this.props;
 
-    const Tag = href ? AnchorStyledButton : StyledButton;
-
-    const buttonLabel = typeof label === 'string' ? (
-      <Text>
-        <strong>{label}</strong>
-      </Text>
-    ) : label;
-
-    const first = reverse ? buttonLabel : icon;
-    const second = reverse ? icon : buttonLabel;
+    let buttonIcon = icon;
+    // only change color if user did not specify the color themselves...
+    if (primary && icon && !icon.props.color) {
```

Comment:
This feels a bit odd, reaching into the props of the icon.

What if we changed to support providing icon as a class, which would allow us to instantiate it with the color we want directly, not needing to clone?

Reply:
not sure I understand your suggestion :(

Reply:
why you think this feels odd ? I think it is a good trade off between the previous solution where we had the `TODO: revisit this`

Reply:
oh I see what you are suggesting. I'm not sure I like that. since you would be forcing people to be using a given structure. Icon today can be really anything, including an svg.

Reply:
But, since it can be anything, is always right to assume what a `color` prop means?

Reply:
I don't think it is always right. But I think we are already relying on that elsewhere. I took this example from FormField where we clone the children to pass `plain`. This is exactly the same thing.

I'm ok with this trade off.